ExtendedWPFToolkitSolution/Src/Xceed.Wpf.Toolkit/NumericUpDown/Implementation/CommonNumericUpDown.cs

```diff
@@ -263,7 +263,7 @@ protected override void OnMaximumChanged( T? oldValue, T? newValue )
         return this.Value;
       }
 
-      var cleanedText = Regex.Replace(text, "[^0-9.,]", "", RegexOptions.Compiled, TimeSpan.FromMilliseconds(100));
+      var cleanedText = Regex.Replace(text, "[^-0-9.,]", "", RegexOptions.Compiled, TimeSpan.FromMilliseconds(100));

```

Comment:
```suggestion
      var match = Regex.Match(FormatString, "^([^\{]*)\{.*\}([^\}]*)$", RegexOptions.Compiled, TimeSpan.FromMilliseconds(100));
      if(match.Success)
      {
          text = text.TrimStart(match.Groups[1].ToString()).TrimEnd(match.Groups[2].ToString());
      }
```
I think we might be better of to parse the format string here and determine which parts of the input text to discard....

Reply:
What would we gain from it?  The regex and its handling code would need to become a lot more complicated, and we know exactly what we're looking for in this control; we don't need anything generalized or extremely flexible.

Reply:
What these fields accept is _broader_ than it needs to be (it'll find a numeric value amidst any arbitrary string and isn't really validating it at all), but I doubt it's worth going too far out of our way for it.

Reply:
Well the logic I suggested should satisfy our needs in getting rid of whatever we add with the formatting string.... I don't see a huge complexity there. I agree we have pretty narrow use cases. However I think the control should be generic enough so that it can handle all our needs. My first attempt (introducing the string cleanup that was there before) obviously failed....

Reply:
Hm, that's a pretty clever regex.  I do like it, and after some experimentation I do think it's a better solution, and it lets code further down decide if the remaining text is valid for the control or not.  I'll incorporate it in and do some testing.